cmake-modules/FolderList.cmake

```diff
@@ -1,23 +1,33 @@
 macro(GetFolderList project)
     message ("project found ${project}")
-  
+    message ("FLAG VALUE : ${FETCH_SOURCE_DEPS}")
     if(${project} STREQUAL CERTIFICATES)
         DownloadDepVersion(sdk/core azure-core 1.2.0)
         DownloadDepVersion(sdk/identity azure-identity 1.1.0)
     elseif(${project} STREQUAL IDENTITY)
         DownloadDepVersion(sdk/core azure-core 1.2.0)
-    endif()
-
-    if(${project} STREQUAL SECRETS)
+    elseif(${project} STREQUAL SECRETS)
         DownloadDepVersion(sdk/core azure-core 1.2.0)
         DownloadDepVersion(sdk/identity azure-identity 1.1.0)
-    endif()
-
-    if(${project} STREQUAL KEYS)
+    elseif(${project} STREQUAL KEYS)
         DownloadDepVersion(sdk/core azure-core 1.2.0)
         DownloadDepVersion(sdk/identity azure-identity 1.1.0)
+    elseif(${project} STREQUAL STORAGE_COMMON)
+        DownloadDepVersion(sdk/core azure-core 1.0.0)
+    elseif(${project} STREQUAL STORAGE_BLOBS)
+        DownloadDepVersion(sdk/core azure-core 1.0.0)
+        DownloadDepVersion(sdk/storage/azure-storage-common azure-storage-common 12.2.0)
+     elseif(${project} STREQUAL STORAGE_DATALAKE)
+        DownloadDepVersion(sdk/core azure-core 1.0.0)
+        DownloadDepVersion(sdk/storage/azure-storage-common azure-storage-common 12.2.0)
+        DownloadDepVersion(sdk/storage/azure-storage-blobs azure-storage-blobs 12.2.0)
+     elseif(${project} STREQUAL STORAGE_SHARES)
+        DownloadDepVersion(sdk/core azure-core 1.0.0)
+        DownloadDepVersion(sdk/storage/azure-storage-common azure-storage-common 12.2.0)
+    elseif(${project} STREQUAL STORAGE_QUEUES)
+        DownloadDepVersion(sdk/core azure-core 1.0.0)
+        DownloadDepVersion(sdk/storage/azure-storage-common azure-storage-common 12.2.0)
```

Comment:
Why do we need this PR? What problem are we trying to fix?

If we have cyclic dependencies(A depends on B and B depends on A), can everything still work?

Now there are two places we have to update when publishing a new version. Can we update this version with a pipeline?

Reply:
we have customers requiring to build from source not using vcpkg. this PR fetches the sources from github for dependencies.

It should, i haven't seen any cyclic ones yet but all builds passed currently. 

since this feature is in development, i was actually thinking of parsing the vcpkg files to get the dependencies. since the versions of the deps don't change that often i left that feature towards the end.